transport/x/grpc/config_test.go

```diff
@@ -210,6 +236,27 @@ func TestTransportSpec(t *testing.T) {
 				inbound, ok := cfg.Inbounds[0].(*Inbound)
 				require.True(t, ok, "expected *Inbound, got %T", cfg.Inbounds[0])
 				assert.Contains(t, inbound.listener.Addr().String(), tt.wantInbound.Address)
+
+				if tt.wantInbound.ServerMaxRecvMsgSize > 0 {
+					assert.Equal(t, tt.wantInbound.ServerMaxRecvMsgSize, inbound.t.options.serverMaxRecvMsgSize)
+				} else {
+					assert.Equal(t, defaultServerMaxRecvMsgSize, inbound.t.options.serverMaxRecvMsgSize)
+				}
+				if tt.wantInbound.ServerMaxSendMsgSize > 0 {
+					assert.Equal(t, tt.wantInbound.ServerMaxSendMsgSize, inbound.t.options.serverMaxSendMsgSize)
+				} else {
+					assert.Equal(t, defaultServerMaxSendMsgSize, inbound.t.options.serverMaxSendMsgSize)
+				}
+				if tt.wantInbound.ClientMaxRecvMsgSize > 0 {
+					assert.Equal(t, tt.wantInbound.ClientMaxRecvMsgSize, inbound.t.options.clientMaxRecvMsgSize)
+				} else {
+					assert.Equal(t, defaultClientMaxRecvMsgSize, inbound.t.options.clientMaxRecvMsgSize)
```

Comment:
optional: Maybe wantInbound.*Size should always be non-zero? So the test cases would explicitly state that they expect `default*Size`.

Reply:
Tried to do this out, actually just makes it dirty in another way, just going to leave for now.